api/allocations.go

```diff
@@ -404,6 +404,36 @@ type NodeScoreMeta struct {
 	NormScore float64
 }
 
+// Stub returns a list stub for the allocation
+func (a *Allocation) Stub() *AllocationListStub {
+	return &AllocationListStub{
+		ID:                    a.ID,
+		EvalID:                a.EvalID,
+		Name:                  a.Name,
+		Namespace:             a.Namespace,
+		NodeID:                a.NodeID,
+		NodeName:              a.NodeName,
+		JobID:                 a.JobID,
+		JobType:               *a.Job.Type,
```

Comment:
This raises a question I don't know the answer to: why do we use `*string` vs `string` in different API structs?

Reply:
We use it to distinguish between fields that were unspecified and intentionally set to `""` (or whatever the zero is). It's a shame because it's hard to know what to replace it with and it defeats most of the off the shelf code generators we could use to keep structs and api synced automatically.

Reply:
Gotcha. Makes sense.